common/src/metrics/mod.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright 2023-, GraphOps and Semiotic Labs.
+// SPDX-License-Identifier: Apache-2.0
+
+use once_cell::sync::Lazy;
+use prometheus::{core::Collector, register, IntCounterVec, Opts, Registry};
+
+#[allow(dead_code)]
+pub static INDEXER_ERROR: Lazy<IntCounterVec> = Lazy::new(|| {
+    let m = IntCounterVec::new(
+        Opts::new("indexer_error", "Indexer errors observed over time")
+            .namespace("indexer")
+            .subsystem("service"),
+        &["code"],
+    )
+    .expect("Failed to create indexer_error");
+    register(Box::new(m.clone())).expect("Failed to register indexer_error counter");
+    m
+});
+
+#[allow(dead_code)]
+pub static REGISTRY: Lazy<Registry> = Lazy::new(Registry::new);
+
+#[allow(dead_code)]
+pub fn register_metrics(registry: &Registry, metrics: Vec<Box<dyn Collector>>) {
+    for metric in metrics {
+        registry.register(metric).expect("Cannot register metrics");
+    }
+}
+
+/// Register indexer error metrics in Prometheus registry
+pub fn register_indexer_error_metrics() {
+    register_metrics(&REGISTRY, vec![Box::new(INDEXER_ERROR.clone())]);
+}
```

Comment:
I think it would be better to just use the default registry instead of creating our own. I don't recall why we're using a custom one in indexer-service (and also graph-node), but I know that the default registry works well with `autometrics` which I believe is used in indexer-service-rs. That's why I think using `register_int_counter_vec!` is probably a good idea, because that uses the default registry.

Reply:
Nice suggestion! I updated the labeling for both `indexer_error` and `query` metrics, used the macros, and removed the custom registry [fb870a7](https://github.com/graphprotocol/indexer-rs/pull/79/commits/fb870a7c649f110504194a15246c2e0a19d69ae1). The code looks a lot cleaner than before.

The current custom registry should work fine with `autometrics`, though we are only using it for `encode_global_metrics` and `global_metrics_exporter`. I think it could be interesting to use `autometrics` for some functions to measure performance, like for `execute_paid_query` and `execute_free_query`. I don't think we have those measurements in TS but do lmk if you think adding some of those within this PR is a good idea